### PR TITLE
CPDRP-889: step by step

### PR DESCRIPTION
**Pull request by ethanmills**
### Context
CPDRP-889

Add a step-by-step page and link to it from the start page

### Changes proposed in this pull request
- Copy loads of styles and horrible JS from elsewhere (mostly govuk publishing components)
- Reduce the horrible-ness just to the point it starts working
- Never touch it again

### Review Checks
- [ ] All pages have automated accessibility checks via cypress
- [ ] All pages have visual tests via cypress + percy
- [ ] All `School` queries are correctly scoped - `eligible` when they need to be

### CODE REVIEW

File: db/seeds/test_data.rb [hunk 1]
```diff
@@ -449,6 +449,8 @@
   mentor_profile.schedule = Finance::Schedule.default
 end
 
-["Capita", "Teach First", "UCL Institute of Education", "Best Practice Network", "Ambition Institute", "Education Development Trust"].each do |provider|
-  ValidTestDataGenerator::LeadProviderPopulater.call(name: provider, total_schools: 10, participants_per_school: 10)
+PaperTrail.request(enabled: false) do
+  ["Capita", "Teach First", "UCL Institute of Education", "Best Practice Network", "Ambition Institute", "Education Development Trust"].each do |provider|
+    ValidTestDataGenerator::LeadProviderPopulater.call(name: provider, total_schools: 10, participants_per_school: 10)
+  end
```
Comment (kicferk1) [inline db/seeds/test_data.rb:455]: We can also reduce it to like 1 school and 5 participants, it's not like we need 100 participants for anything.

File: app/webpacker/styles/components/_step-by-step-header.scss [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// Based on https://github.com/alphagov/govuk_publishing_components/blob/v22.0.0/app/assets/stylesheets/govuk_publishing_components/components/_step-by-step-nav-header.scss
```
Comment (kicferk1) [inline app/webpacker/styles/components/_step-by-step-header.scss:1]: Can we put it in `vendor`, like we do with nhs table?

File: app/views/step_by_step/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+    <p class="govuk-body">From September 2021, statutory induction for early career teachers (also known as NQTs) will change as part of the early career framework (ECF) reforms. <a href="https://www.gov.uk/government/publications/early-career-framework-reforms-overview/early-career-framework-reforms-overview" class="govuk-link">Read more about the reforms in our guidance</a>.</p>
+    <p class="govuk-body">All early career teachers (ECTs) are now entitled to a funded 2-year induction programme. This includes structured training and support to help them get their teaching career off to the best possible start.</p>
+    <p class="govuk-body">Your school still needs to <a href="https://www.gov.uk/government/publications/early-career-framework-reforms-overview/early-career-framework-reforms-overview#the-role-of-the-appropriate-body" class="govuk-link">have an appropriate body in place</a>.</p>
+
+    <h2 class="govuk-heading-m">When to set up your school’s programme</h2>
+    <p class="govuk-body">Any school that offers statutory induction needs to follow the steps below as soon as possible.</p>
+
+
```
Comment (kicferk1) [inline app/views/step_by_step/show.html.erb:14]: Evil whitespace

File: app/views/step_by_step/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+              					<span class="app-step-nav__circle app-step-nav__circle--number">
+                					<span class="app-step-nav__circle-inner">
+                  						<span class="app-step-nav__circle-background">
+                    						<span class="govuk-visually-hidden">Step</span> 1
+                  						</span>
+                					</span>
+              					</span>
+              <span class="js-step-title">Tell us whether or not you expect ECTs to join you in this academic year</span>
+            </h2>
+          </div>
```
Comment (kicferk1) [inline app/views/step_by_step/show.html.erb:30]: Spans are a bit off

File: app/views/step_by_step/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+
+          <div class="app-step-nav__panel js-panel js-hidden" id="step-panel-tell-us-if-you-have-ects-1">
+            <p class="app-step-nav__paragraph">We recommend you do this as soon as possible, even if you’re not sure.</p>
+
+
```
Comment (kicferk1) [inline app/views/step_by_step/show.html.erb:35]: Evil whitespace

File: app/views/step_by_step/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+            <h2 class="app-step-nav__title">
+              					<span class="app-step-nav__circle app-step-nav__circle--number">
+                					<span class="app-step-nav__circle-inner">
+                  						<span class="app-step-nav__circle-background">
+                    						<span class="govuk-visually-hidden">Step</span> 2
+                  						</span>
+                					</span>
+              					</span>
+              <span class="js-step-title">Nominate an induction tutor to manage your school’s programme</span>
+            </h2>
```
Comment (kicferk1) [inline app/views/step_by_step/show.html.erb:53]: Naughty spans

File: app/views/step_by_step/show.html.erb [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+            <h2 class="app-step-nav__title">
+              					<span class="app-step-nav__circle app-step-nav__circle--number">
+                					<span class="app-step-nav__circle-inner">
+                  						<span class="app-step-nav__circle-background">
+                    						<span class="govuk-visually-hidden">Step</span> 3
+                  						</span>
+                					</span>
+              					</span>
+              <span class="js-step-title">Choose how to run your programme</span>
+            </h2>
```
Comment (kicferk1) [inline app/views/step_by_step/show.html.erb:73]: More naughty spans

File: app/views/step_by_step/show.html.erb [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+              <span class="js-step-title">Add ECTs and their mentors to your training cohort</span>
+            </h2>
+          </div>
+          <div class="app-step-nav__panel js-panel js-hidden" id="step-panel-add-ects-and-mentors-6">
+            <p class="app-step-nav__paragraph">Your induction tutor needs to tell us each ECTs name and email address as soon as possible. We also need this contact information for the mentors you've assigned to work with them. </p>
+            <p class="app-step-nav__paragraph">We’ll contact each ECT and mentor to ask them for more information. We’ll use this to check they’re eligible for this funded training and support.</p>
+            <ol class="app-step-nav__list " data-length="3">
+              <li class="app-step-nav__list-item js-list-item "><%=govuk_link_to "Tell us about your 2021 to 2022 ECTs and mentors ", root_path, class: "app-step-nav__link js-link" %></li>
+
+            </ol>
```
Comment (kicferk1) [inline app/views/step_by_step/show.html.erb:150]: The empty line looks a bit weird

File: app/views/step_by_step/show.html.erb [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+            <h3 class="govuk-heading-s govuk-!-margin-bottom-1 govuk-!-padding-top-1">If you’re using a DfE-funded training provider</h3>
+            <p class="app-step-nav__paragraph">They’ll contact you directly about what to do next.</p>
+
+            <h3 class="govuk-heading-s govuk-!-margin-bottom-1 govuk-!-padding-top-1">If you’re using DfE-accredited materials to deliver your own programme</h3>
+            <p class="app-step-nav__paragraph">We’ll send your induction tutor, ECTs and mentors a link to access the materials you choose.</p>
+
+            <h3 class="govuk-heading-s govuk-!-margin-bottom-1 govuk-!-padding-top-1">If you’re designing and delivering your own programme</h3>
+            <p class="app-step-nav__paragraph">You need to plan and create your materials.</p>
+
+
```
Comment (kicferk1) [inline app/views/step_by_step/show.html.erb:181]: Evil whitespace

File: app/webpacker/packs/components/step-by-step-nav.js [hunk 10]
```diff
@@ -0,0 +1,476 @@
+// Adapted from https://raw.githubusercontent.com/alphagov/govuk_publishing_components/master/app/assets/javascripts/govuk_publishing_components/components/step-by-step-nav.js
+
+/* eslint-disable */
```
Comment (kicferk1) [inline app/webpacker/packs/components/step-by-step-nav.js:3]: Hahaha, I love that we just disable linter for all of that 😂

Just to be sure... Our govuk-components don't have this particular component, right?

**Review by kicferk1 — COMMENTED** [review]
Only real comments are about sanitizing the long html file

File: app/webpacker/packs/components/step-by-step-nav.js (hunk 10 above)
Comment (ethanmills) [reply app/webpacker/packs/components/step-by-step-nav.js:3]: No, this generally lives on a top level gov.uk page, so isn't included anywhere. Publishing components obviously has it, but I really don't want to jump through all the hoops to use that...

**Review by kicferk1 — APPROVED** [review]

**Pull request merged**